azure-ipam/logger/logger.go

```diff
@@ -6,56 +6,48 @@ import (
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 type Config struct {
-	Level            string
-	OutputPaths      string // comma separated list of paths
-	ErrorOutputPaths string // comma separated list of paths
+	Level       string // Debug by default
+	Filename   	string
+	MaxSize    	int    // megabytes
+	MaxBackups 	int    // # of backups, no limitation by default
+	MaxAge     	int    // days, no limitation by default
 }
 
 // NewLogger creates and returns a zap logger and a clean up function
 func New(cfg *Config) (*zap.Logger, func(), error) {
-	loggerCfg := &zap.Config{}
-
-	level, err := zapcore.ParseLevel(cfg.Level)
-	if err != nil {
-		return nil, nil, errors.Wrapf(err, "failed to parse log level")
-	}
-	loggerCfg.Level = zap.NewAtomicLevelAt(level)
-
-	loggerCfg.Encoding = "json"
-	loggerCfg.OutputPaths = getLogOutputPath(cfg.OutputPaths)
-	loggerCfg.ErrorOutputPaths = getErrOutputPath(cfg.ErrorOutputPaths)
-	loggerCfg.EncoderConfig = zapcore.EncoderConfig{
-		TimeKey:     "time",
-		MessageKey:  "msg",
-		LevelKey:    "level",
-		EncodeLevel: zapcore.LowercaseLevelEncoder,
-		EncodeTime:  zapcore.ISO8601TimeEncoder,
-	}
-
-	logger, err := loggerCfg.Build()
-	if err != nil {
+	//check the filepath is not empty
+	if cfg.Filename == "" {
+		err := errors.New("no Filename is provided")
 		return nil, nil, errors.Wrapf(err, "failed to build zap logger")
 	}
-
+	// set the log level - Debug by default
+	var logLevel zapcore.Level
+	if strings.ToLower(cfg.Level) == "info" {
+		logLevel = zap.InfoLevel
+	} else if strings.ToLower(cfg.Level) == "error" {
+		logLevel = zap.ErrorLevel
+	} else {
+		logLevel = zap.DebugLevel
+	}
```

Comment:
imo [this was previously done correctly](https://github.com/Azure/azure-container-networking/pull/1523/files#diff-00339433bfed6e3f6830a7e8f1ecd201d726875a0313512452c04a20a8bc8c6fL21-L25) as 
```go
level, err := zapcore.ParseLevel(cfg.Level)
```
what is the reason to change this to parse the strings ourselves?

Reply:
The issue that I had with the parseLevel was that if you pass "" as level or any of the non-defined string, it will through an Error and apparently prevent Azure-IPAM execution (That's what I observed). @tamilmani1989 suggested to have Debug mode as default for Azure-IPAM. 
I can still use ParseLevel() but if an error occurs, just set the level as debug instead of returning an error.

Reply:
@AzureAhai  lets not change this and use ParseLevel itself and debug is defined as default in main.go already